### PR TITLE
Server polling

### PR DESCRIPTION
**Pull request by stephen-lt-sl**
@stevenhillcox @sievins @jebbinSCL

### CODE REVIEW

File: public/main.js [hunk 1]
```diff
@@ -124,6 +131,7 @@ function reloadTodoList() {
         } else {
             clearCompleteButton.style.visibility = "hidden";
         }
+        todoListRefreshTimerId = setTimeout(reloadTodoList, todoListRefreshRate);
```
Comment (stevenhillcox-sl) [inline public/main.js]: Your code works, but there is a nicer way of doing this.
You should look into the javascript function setInterval().

Comment (stephen-lt-sl) [inline public/main.js]: That was my original plan; the advantage of doing it this way is that the refresh timer is reset by the call to "clearTimeout" at the top of reloadTodoList, and the next timer is only set when the todo list finishes refreshing. This means that the client won't try to get the list again right after (or during) a different refresh of the list, such as when the list is changed by some user action or a particularly slow previous timed refresh.

Comment (stevenhillcox-sl) [inline public/main.js]: If you are going with that route, I am not sure why you clear the timeout on line 94.
I would still suggest you use setInterval. 4 seconds is more than enough time for the server to respond and for your page to update itself. 
If you want to cover failure cases, I would catch any non 200 response from the server and clear the interval at that point.

**Comment by sievins** [comment]
:shipit:

**Pull request merged**